components/directory-server-manager/org.wso2.carbon.directory.server.manager.ui/src/main/resources/web/servicestore/add-finish.jsp

```diff
@@ -84,14 +84,14 @@
     if (qpplicationComponentFound) {
         if (!isError) {
     %>
-    location.href = '../application/configure-service-provider.jsp?action=update&display=kerberos&spName=<%=spName%>&kerberos=<%=servicePrincipleName%>';
+    location.href = '../application/configure-service-provider.jsp?action=update&display=kerberos&spName=<%=Encode.forJavaScriptBlock(Encode.forUriComponent(spName))%>&kerberos=<%=Encode.forJavaScriptBlock(Encode.forUriComponent(servicePrincipleName))%>';
     <%  } else { %>
-    location.href = '../application/configure-service-provider.jsp?action=cancel&display=kerberos&spName=<%=spName%>';
+    location.href = '../application/configure-service-provider.jsp?action=cancel&display=kerberos&spName=<%=Encode.forJavaScriptBlock(Encode.forUriComponent(spName))%>';
     <%
         }
     }else {
     %>
-    location.href = '<%=forwardTo%>';
+    location.href = '<%=Encode.forJavaScriptBlock(forwardTo)%>';
```

Comment:
forwardTo is assigned hardcoded values which does not contain special characters. So it's in trusted scope. Not necessary to encode here